### ISSUE TITLE
No store type when using module().child(...)  as per README

### ISSUE THREAD

**Issue by f-matos** [ktsn/sinai#24]
Hi,

When I try to reproduce the code on the README, I get `Property 'counter' does not exist on type '{}'`  error when doing `store.state.counter.count`

The store declaration is

```typescript
// Create root module
const root = module().child('counter', counter)

// Create store
const rootStore = store(root, {
  strict: process.env.NODE_ENV !== 'production'
})
```

But if I pass the counter module instead, it works as expected when doing `store.state.count` :

```typescript
// Create store
const rootStore = store(counter, { // <--
  strict: process.env.NODE_ENV !== 'production'
})
```

and I couldn't figure out how to make "module()" work. Is there something I'm missing? What I want is to partition my store in several modules.

**Comment by ktsn** [comment]
I just fixed the example. https://github.com/ktsn/sinai/commit/3e5981e043e66c4f2edd0858c0694040cb545732

**Comment by f-matos** [comment]
Oh I feel... not so clever right now.
I had noticed the duplicated `store` naming problem and had fixed it, but was still was having the same problem.
The *actual* problem was that I copied the hot reload thing from `sinai/example/store/index.ts`

```
declare const module: any
if (module.hot) {
  module.hot.accept(['./counter'], () => {
    const newModule = require('./counter').default
    s.hotUpdate(newModule)
  })
}
```

But this obviously causes a problem when doing `const root = module().child('counter', counter)`. So that was that.  Thanks for the help, `sinai` is a great library, I'm having a lot of fun with it.
